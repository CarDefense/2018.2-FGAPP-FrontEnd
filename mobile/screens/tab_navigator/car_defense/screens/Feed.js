import React from 'react';
import { FlatList, Text, View, StyleSheet, ScrollView, RefreshControl, Image, TouchableOpacity } from 'react-native';
import { NOTIFICATIONS_API, PROFILE_API } from './TabNavigator/const/Const.js'
import Expo from 'expo'

var tk

async function register() {
  const { status } = await Expo.Permissions.askAsync(
    Expo.Permissions.NOTIFICATIONS
  );
  if (status != 'granted') {
    alert('You need to enable permissions in settings');
    return;
  }

  const value = await Expo.Notifications.getExpoPushTokenAsync();
  tk = value;
  //console.log(status, value);

}

export default class Feed extends React.Component {
  componentWillMount() {
    register();
    this.listener = Expo.Notifications.addListener(this.listen);
  }
  componentWillUnmount() {
    this.listener && Expo.Notifications.addListener(this.listen);
  }

  listen = ({ origin, data }) => {
    console.log('cool data', origin, data);
  }

  constructor(props) {
    super(props);
    this.state = {
      refreshing: false,
      height: 135
    }
  }

  async componentDidMount() {
    const { state } = this.props.navigation;
    var id = state.params ? (state.params.user.id ? state.params.user.id : state.params.user.user_id) : undefined;

    let profile = JSON.stringify({
      id_token: id,
      notification_token: tk,
    })
    //console.log(profile);
    fetch(PROFILE_API + '/set_token/', {
      method: 'POST',
      headers: {
        'Accept': 'application/json',
        'Content-Type': 'application/json',
      },
      body: profile
    }).then(response => { return response.json() }
    ).then(jsonResponse => {
      //console.log(jsonResponse);
    }
    ).catch(error => {
      //console.log(error)
    })

    return fetch(NOTIFICATIONS_API + '/emergencynotifications/?ordering=-id')
      .then((response) => response.json())
      .then((responseJson) => {

        this.setState({
          dataSource: responseJson,
        }, function () {

        });

      })
      .catch((error) => {
        //console.error(error);
      });
  }

  _onRefresh = () => {
    this.setState({ refreshing: true });
    this.componentDidMount().then(() => {
      this.setState({ refreshing: false });
    });
  }

  _height = (height) => {
    if (height === 400) {
      this.state.height = 135
      this.setState({ refreshing: true });
      this.componentDidMount().then(() => {
        this.setState({ refreshing: false });
      });
    }
    else {
      this.state.height = 400
      this.setState({ refreshing: true });
      this.componentDidMount().then(() => {
        this.setState({ refreshing: false });
      });
    }
  }

  render() {

    return (
      <View style={{ backgroundColor: '#00ACC1', flex: 1 }}>
        <ScrollView 
          style={styles.item}
          refreshControl={
            <RefreshControl
              refreshing={this.state.refreshing}
              onRefresh={this._onRefresh}
            />
          }
        >

          <FlatList
            data={this.state.dataSource}
            renderItem={({ item }) => {
              return (
                <View style={styles.item2}>
                  <Text style={styles.text1}>{item.title}</Text>
                  <Text style={styles.text}>{item.date} Ã s {item.time}</Text>
                  <TouchableOpacity
                    onPress={() => { this._height(this.state.height) }}
                  >
                    <Image source={{ uri: item.image }}
                      style={{ height: this.state.height }} />
                  </TouchableOpacity>
                  <View style={{ marginTop: 10 }}>
                    <Text style={styles.text2}>{item.message}</Text>
                  </View>
                </View>
              );
            }}
            keyExtractor={({ id }, index) => id.toString()}

          />

        </ScrollView>
      </View >

    );
  }
}

const styles = StyleSheet.create({
  item: {
    shadowColor: "#000000",
    shadowOpacity: 0.8,
    shadowRadius: 2,
    shadowOffset: {
      height: 1,
      width: 1
    },
    elevation: 4,
    marginTop: 20,
    marginBottom: 5
  },

  text: {
    color: "#B2EBF2",
    fontWeight: '600'
  },
  headerContent: {
    padding: 30,
    alignItems: 'center',
  },
  avatar: {
    width: 100,
    height: 100,
    borderRadius: 63,
    borderWidth: 4,
    borderColor: "white",
    marginBottom: 10,
  },
  name: {
    fontSize: 22,
    color: "white",
    fontWeight: '800',
  },
  userInfo: {
    fontSize: 15,
    color: "white",
    fontWeight: '800',
    marginTop: 30,
    marginBottom: 15
  },

  icon: {
    width: 30,
    height: 30,
    marginTop: 20,
  },
  info: {
    fontSize: 18,
    marginTop: 20,
    color: "#FFFFFF",
  },
  item2: {
    backgroundColor: "white",
    flexGrow: 1,
    padding: 20,
    borderRadius: 15,
    elevation: 4,
    margin: 25,
    marginTop: 2,

  },
  icon1: {
    color: "#B2EBF2",
    fontWeight: '800',
    fontSize: 44,
    position: 'absolute',
    left: 10,
    marginTop: 5,
    bottom: 25// Keep some space between your left border and Image

  },
  text1: {
    color: "#26C6DA",
    fontWeight: '800',
    fontSize: 30
  },
  text2: {
    color: "#26C6DA",
    fontWeight: '800',
  }
});