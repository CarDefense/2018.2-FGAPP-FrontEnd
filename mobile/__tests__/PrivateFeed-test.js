import Feed from '../screens/tab_navigator/car_defense/screens/TabNavigator/PrivateFeed/PrivateFeed'
import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'

test('PrivateFeed snapShot', () => {
    const snap = renderer.create(
        <Feed/>
    ).toJSON()
    expect(snap).toMatchSnapshot()
})