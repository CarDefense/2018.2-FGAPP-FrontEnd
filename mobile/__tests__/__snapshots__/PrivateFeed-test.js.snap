// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrivateFeed snapShot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#00ACC1",
      "flex": 1,
    }
  }
>
  <RCTScrollView>
    <View>
      <View
        style={
          Object {
            "backgroundColor": "white",
            "borderRadius": 15,
            "elevation": 4,
            "flexGrow": 1,
            "margin": 25,
            "marginTop": 20,
            "padding": 20,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#26C6DA",
              "fontSize": 30,
              "fontWeight": "800",
            }
          }
        >
          Olá 
          
          , seja bem Vindo!
        </Text>
        <View
          style={
            Object {
              "marginTop": 10,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#26C6DA",
                "fontWeight": "800",
              }
            }
          >
            Cadastre a placa do seu veículo para poder receber novas notificações.
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "paddingTop": "75%",
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              Object {
                "alignItems": "center",
                "backgroundColor": "rgba(0,0,0,0)",
                "justifyContent": "center",
              },
            ]
          }
        >
          <ActivityIndicator
            animating={true}
            color="white"
            hidesWhenStopped={true}
            size={60}
          />
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
